hbase-server/src/main/java/org/apache/hadoop/hbase/backup/HFileArchiver.java

```diff
@@ -439,7 +439,7 @@ private static List<File> resolveAndArchive(FileSystem fs, Path baseArchiveDir,
 
     List<File> failures = new ArrayList<>();
     String startTime = Long.toString(start);
-    for (File file : toArchive) {
+    toArchive.parallelStream().forEach(file -> {
```

Comment:
Also if toArchive is small,  there are chances that making it parallel can take more time than doing the same sequentially.

Reply:
This wouldn't be the case as it is an IO operation so parallelizing it would improve it over what we already have.

Reply:
I thought `parallelStream` (and its underlying `ForkJoinPool`) is not really for I/O blocking operations.

Reply:
Yes, that's another point to consider